src/tests/ftest/server/metadata.py

```diff
@@ -278,6 +278,17 @@ def test_metadata_fillup(self):
                 self.fail("Phase 3: fail (unexpected container create error)")
         self.log.info(
             "Phase 3: passed (created %d / %d containers)", len(self.container), loop)
+
+        # Phase 4 destroy pool while metadata rdb is full
+        # And prevent container destroy steps from running in teardown
+        self.log.info("Phase 4: Destroy pool while metadata rdb is full (expected to work)")
+        self.container = None
+        try:
+            self.destroy_pools(self.pool)
+            self.log.info("Phase 4: passed (pool with full metadata rdb successfully destroyed)")
+        except TestFail as error:
+            self.log.error(str(error))
+            self.fail("Phase 4: fail (pool with full metadata rdb NOT destroyed)")
```

Comment:
The `destroy_pools` method reports any raised exceptions in the list of errors it returns.
```suggestion
        destroy_errors = remove_pool(self, self.pool)
        for error in destroy_errors:
            self.log.error(error)
        if destroy_errors:
            self.fail("Phase 4: fail (pool with full metadata rdb NOT destroyed)")
        self.log.info("Phase 4: passed (pool with full metadata rdb successfully destroyed)")
```

This will also require a `from test_utils_pool import remove_pool` import.

Reply:
Thanks for taking another look at this. We also discussed in a chat that another solution that keeps the pool destroy in the usual teardown flow could be achieved by simply setting self.contianer = None here in the test.

Before pushing a final version of this patch, I'll wait for PR https://github.com/daos-stack/daos/pull/10479 to land (that will address the 2 test failures seen while testing this PR).